src/main.rs

```diff
@@ -109,3 +117,110 @@ pub fn get_zones(pbf: &mut OsmPbfReader) {
         }
     }
 }
+
+pub fn read_libpostal_yaml_folder(yaml_files_folder: String) -> BTreeMap<String, model::Country> {
+    let paths = fs::read_dir(yaml_files_folder).expect("The yaml directory doesn't exist.");
+    let mut countries: BTreeMap<String, model::Country> = BTreeMap::new();
+
+    for entry in paths {
+        let mut contents = String::new();
+        let a_path = entry.unwrap().path();
+
+        File::open(&a_path)
+            .unwrap()
+            .read_to_string(&mut contents)
+            .expect("Something went wrong reading the file");
+
+        let deserialized_country = match read_libpostal_yaml(&contents) {
+            Ok(a) => a,
+            Err(_) => continue,
+        };
+
+        let reference = a_path
+            .file_name()
+            .unwrap()
+            .to_str()
+            .unwrap()
+            .get(0..2)
+            .unwrap();
+        info!("{:?}: {:?}", reference, deserialized_country);
+        if !deserialized_country.admin_level.contains_key("error") {
+            countries.insert(reference.into(), deserialized_country);
+        }
+    }
+    countries
+}
+
+pub fn read_libpostal_yaml(contents: &String) -> Result<model::Country, String> {
```

Comment:
it's not the way to handle errors, you don't use the Result here.

I think you can write this:

```
fn read_libpostal_yaml(contents: &String) -> Result<model::Country, Error> {
    serde_yaml::from_str(&contents)?
}
```
:stuck_out_tongue_winking_eye: 

(if you have the failure crate)